### PR TITLE
Add Managed Service with Serverless post

### PR DESCRIPTION
**Pull request by hasheddan**
"In this post Dan from Crossplane explains how to easily deploy and consume managed services alongside your serverless functions on Kubernetes using Crossplane and OpenFaaS."

I have run the blog locally with `docker-compose up` and everything appears to be in order except for the header image is not formatted well, any insight on how we can either crop it or apply CSS in the front matter?

cc @alexellis 

header-image
 

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

### CODE REVIEW

**Comment by alexellis** [comment]
Can you try to scale to the viewport?

**Comment by alexellis** [comment]
We did similar here with a plainer background image: https://www.openfaas.com/blog/birthday-teamserverless/

Okteto tried to embed a complex graphic behind the title text and it wasn't particularly readable: https://www.openfaas.com/blog/painless-serverless-development/

**Comment by alexellis** [comment]
Some of your images are over 1MB in size, some over 2MB. This could result in 10-15MB + of downloads just to view this page which is not ideal from a bandwidth point of view. Can you scale them back or compress?

**Comment by hasheddan** [comment]
@alexellis I have updated the cover photo to something that is less complex and have compressed all of the images :) 

openfaas-cover-2

File: Gemfile.lock [hunk 1]
```diff
@@ -259,8 +259,6 @@ DEPENDENCIES
   github-pages
   html-proofer
   jekyll-archives
-  jekyll-paginate
-  jekyll-sitemap
```
Comment (alexellis) [inline Gemfile.lock]: Did you mean to remove both of those Gems?

Comment (hasheddan) [reply Gemfile.lock]: Hmm no, I think it happened when running locally. Let me revert that change :+1:

**Comment by alexellis** [comment]
Thanks for compressing however you've added the uncompressed versions to history, and then added compressed on top meaning they are still going to bloat the repo. Could you squash the two commits with `git rebase -i HEAD~2`?

**Comment by alexellis** [comment]
Then because of `Signed-off-by: hasheddan <georgedanielmangum@gmail.com>`, please run:

```
git config user.name Daniel Mangum
```

Then:

```
git commit -s --amend
```

Thanks 👍

File: _posts/2020-3-29-managed-services-serverless.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+<img src="/images/2020-03-30-managed-services-serverless/mangum.jpg" width="25%" height="25%" />
+
+Kubernetes is often seen as complex and confusing for new and seasoned users alike. While it provides a wealth of tools and functionality for infrastructure operators, developers frequently struggle to navigate the complexity to just get their application running. OpenFaaS eases that burden by providing the fastest way to get your code running on Kubernetes.
+
+However, while OpenFaaS addresses the problem of getting your code to production, providing data for your functions is a different beast entirely. Most applications do not provide much value without data, so this is vital piece of the puzzle.
```
Comment (alexellis) [inline _posts/2020-3-29-managed-services-serverless.md]: `Most applications do not provide much value without data, so this is vital piece of the puzzle.` < disagree.

Please reframe in a more positive connotation.

Stateless can be very effective in a number of important use-cases.

File: _posts/2020-3-29-managed-services-serverless.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+- **Infrastructure as Code / APIs**: Nearly every cloud provider supplies an API and set of SDKs that are used by infrastructure as code tools to make it possible to version control your infrastructure provisioning and configuration. This is a *massive* improvement over manual, ad-hoc provisioning, but typically requires a user executing command from their local machine, or manually editing detailed configuration every time they want a new service deployed. Once again, this process is also decoupled from deploying your functions on OpenFaaS.
+- **Running Data Services on Kubernetes Directly**: This path brings you closer to managing both your application code and data services with the same tools and workflow. However, it requires that you shun the high SLAs of experienced cloud providers in favor of administrating complex databases, caches, and queues yourself. This is generally a suitable option for large, experienced organizations, or applications that are not mission critical, but can impose significant burden on companies that are not in the business of running data services.
```
Comment (alexellis) [inline _posts/2020-3-29-managed-services-serverless.md]: I would like to see you mention other alternatives to Crossplane, such as Service Catalog via the Open Source Broker API

https://kubernetes.io/docs/concepts/extend-kubernetes/service-catalog/

File: _posts/2020-3-29-managed-services-serverless.md (hunk 2 above)
Comment (hasheddan) [reply _posts/2020-3-29-managed-services-serverless.md]: I think just removing that sentence would be appropriate here

File: _posts/2020-3-29-managed-services-serverless.md (hunk 3 above)
Comment (hasheddan) [reply _posts/2020-3-29-managed-services-serverless.md]: Hmm I would be happy to mention that the Kubernetes Service Catalog was an attempt to achieve similar functionality, but I don't think that it is very actively maintained at this point.

Comment (hasheddan) [reply _posts/2020-3-29-managed-services-serverless.md]: I have made the other updates you have requested :+1:

Comment (alexellis) [reply _posts/2020-3-29-managed-services-serverless.md]: OSB Open Source Broker API is used extensively in CloudFoundry

Comment (hasheddan) [reply _posts/2020-3-29-managed-services-serverless.md]: Not sure how to introduce this. It feels out of place to add something about CloudFoundry in this post that focuses on Kubernetes. Any thoughts?

File: _posts/2020-3-29-managed-services-serverless.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+> Author bio: [Dan Mangum](https://twitter.com/hasheddan) is a software engineer at [Upbound](https://upbound.io/) where he works on the open source [Crossplane](https://crossplane.io/) project. He also serves on the Kubernetes release team, and is an active contributor to the Kubernetes project and multiple other open source efforts. He hosts a biweekly live stream show, [The Binding Status](https://www.youtube.com/playlist?list=PL510POnNVaaYFuK-B_SIUrpIonCtLVOzT), focused on extending Kubernetes, building Crossplane, and shaping the future of cloud-native computing.
+
+<img src="/images/2020-03-30-managed-services-serverless/mangum.jpg" width="25%" height="25%" />
+
+Kubernetes is often seen as complex and confusing for new and seasoned users alike. While it provides a wealth of tools and functionality for infrastructure operators, developers frequently struggle to navigate the complexity to just get their application running. OpenFaaS eases that burden by providing the fastest way to get your code running on Kubernetes.
+
+However, while OpenFaaS addresses the problem of getting your code to production, providing data for your functions is a different beast entirely.
```
Comment (alexellis) [inline _posts/2020-3-29-managed-services-serverless.md]: I like it, thank you.

**Comment by alexellis** [comment]
I just ran this locally and it rendered as expected.

Please could you add the youtube video and a final sub-heading before the end of the article? You'll see examples of this in the 3rd birthday post. I don't mind this as an additive commit.

**Comment by hasheddan** [comment]
@alexellis do you mean an OpenFaaS `Join Our Community` section or just a `Join Our Community` heading prior to the Crossplane links?

**Comment by alexellis** [comment]
No, I mean I'd like you to embed the video in its own sub-heading. Sorry for not being clear. The embed code is in the 3rd birthday blog post for how to use the syntax in mkdocs.

**Comment by hasheddan** [comment]
@alexellis no worries! Done :+1:

File: _posts/2020-3-29-managed-services-serverless.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+We finished up by sharing the issue link the the live stream audience, then [watched to see](https://youtu.be/XphQgB87U-s?t=3957) which emoji rose to the top.
+
+![View Emojis](/images/2020-03-30-managed-services-serverless/view-emojis.png)
+
+# Watch the Live Stream
+
+If you enjoyed these demos, watch the full live stream to see them in action and learn more about OpenFaaS and Crossplane.
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/XphQgB87U-s" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (alexellis) [inline _posts/2020-3-29-managed-services-serverless.md]: This isn't the format. See https://raw.githubusercontent.com/openfaas/openfaas.github.io/master/_posts/2020-02-14-swag-store-launch.md

i.e.

```
{% include youtube.html id="SzKbMXBiv7s" %}
```

File: _posts/2020-3-29-managed-services-serverless.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+![Inlets Client](/images/2020-03-30-managed-services-serverless/inlets-client.png)
+
+We finished up by sharing the issue link the the live stream audience, then [watched to see](https://youtu.be/XphQgB87U-s?t=3957) which emoji rose to the top.
+
+![View Emojis](/images/2020-03-30-managed-services-serverless/view-emojis.png)
+
+# Watch the Live Stream
+
+If you enjoyed these demos, watch the full live stream to see them in action and learn more about OpenFaaS and Crossplane.
```
Comment (alexellis) [inline _posts/2020-3-29-managed-services-serverless.md]: You could also mention that you interviewed me on openfaas and note one or two questions you asked.

Something like: "You'll also hear me interview Alex at the start of the session where I ask questions like... " Why? It's a more engaging reason to click than "watch the full version"

File: _posts/2020-3-29-managed-services-serverless.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+  engineVersion: "11"
+```
+
+After creating both the `CloudSQLInstanceClass` and the `PostgreSQLInstance` claim, GCP began to provision the database. This [was reflected](https://youtu.be/XphQgB87U-s?t=3190) in the state of the `CloudSQLInstance` object in our Kubernetes cluster, as it reported back `PENDING_CREATE`:
+
+![GCP Pending Create](/images/2020-03-30-managed-services-serverless/gcp-pending-create.png)
+
+After a short period of time, the Postgres database was fully provisioned and the connection information was propagated to the `openfaas-fn` namespace in the form of a Kubernetes `Secret`. We used the connection information to [setup our database and table schema](https://youtu.be/XphQgB87U-s?t=3628) prior to deploying our functions.
+
+```console
```
Comment (alexellis) [inline _posts/2020-3-29-managed-services-serverless.md]: Does that render correctly? I think we use ```bash everywhere else

File: _posts/2020-3-29-managed-services-serverless.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+![View Emojis](/images/2020-03-30-managed-services-serverless/view-emojis.png)
+
+# Watch the Live Stream
+
+If you enjoyed these demos, watch the full live stream to see them in action and learn more about OpenFaaS and Crossplane.
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/XphQgB87U-s" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+# Wrapping Up
+
```
Comment (alexellis) [inline _posts/2020-3-29-managed-services-serverless.md]: You mentioned 3 considerations / value-adds of Crossplane - Separation of concerns etc., here would be a good place to recap them in a brief sentence and to call back to that statement. Under "ideal world"

File: _posts/2020-3-29-managed-services-serverless.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+Bringing together Crossplane and OpenFaaS creates the easiest path for deploying both your code and your managed services on Kubernetes. These two demos show interesting use-cases, but the beautiful thing about both projects is that they are built to be extensible. Whether you are building new OpenFaaS [function triggers](https://docs.openfaas.com/reference/triggers/), adding support for a [new managed service](https://crossplane.io/docs/v0.9/contributing/services_developer_guide.html) on Crossplane, or just bringing the two projects together to get your application up and running, feel free to [reach out](https://twitter.com/hasheddan) to me directly.
+
+Otherwise, bring your thoughts, questions, and feedback to one of our many forums:
```
Comment (alexellis) [inline _posts/2020-3-29-managed-services-serverless.md]: Shouldn't OpenFaaS slack be here too?

File: _posts/2020-3-29-managed-services-serverless.md (hunk 7 above)
Comment (hasheddan) [reply _posts/2020-3-29-managed-services-serverless.md]: Used `console` here because Okteto post did. Seems to render fine

**Comment by hasheddan** [comment]
Thanks for the feedback @alexellis :) I have updated based on your edits!

**Comment by alexellis** [comment]
Merging, thank you for the putting this together so quickly. It'll be up on the site in a few moments.

**Pull request merged**